x-pack/plugins/ml/public/timeseriesexplorer/components/timeseries_chart/timeseries_chart.js

```diff
@@ -75,25 +88,116 @@ function getSvgHeight() {
 
 export class TimeseriesChart extends React.Component {
   static propTypes = {
+    indexAnnotation: PropTypes.func,
     autoZoomDuration: PropTypes.number,
     contextAggregationInterval: PropTypes.object,
     contextChartData: PropTypes.array,
     contextForecastData: PropTypes.array,
     contextChartSelected: PropTypes.func.isRequired,
+    deleteAnnotation: PropTypes.func,
     detectorIndex: PropTypes.string,
     focusAggregationInterval: PropTypes.object,
+    focusAnnotationData: PropTypes.array,
     focusChartData: PropTypes.array,
     focusForecastData: PropTypes.array,
     modelPlotEnabled: PropTypes.bool.isRequired,
+    refresh: PropTypes.func,
     renderFocusChartOnly: PropTypes.bool.isRequired,
     selectedJob: PropTypes.object,
     showForecast: PropTypes.bool.isRequired,
     showModelBounds: PropTypes.bool.isRequired,
     svgWidth: PropTypes.number.isRequired,
     swimlaneData: PropTypes.array,
     timefilter: PropTypes.object.isRequired,
+    toastNotifications: PropTypes.object,
     zoomFrom: PropTypes.object,
     zoomTo: PropTypes.object
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      annotation: {},
+      isFlyoutVisible: false,
+      isSwitchChecked: true,
+    };
+
+    this.closeFlyout = this.closeFlyout.bind(this);
+    this.showFlyout = this.showFlyout.bind(this);
+  }
+
+  onSwitchChange = () => {
```

Comment:
I'm not seeing this being used anywhere 🤔

Reply:
Good catch! Copy-paste oversight :) Fixed in fb52cccbf9